src/main/java/org/folio/helper/PurchaseOrderHelper.java

```diff
@@ -273,25 +273,17 @@ public Future<Void> updateOrder(CompositePurchaseOrder compPO, boolean deleteHol
   public Future<Void> handleFinalOrderStatus(CompositePurchaseOrder compPO, String initialOrdersStatus,
                                                         RequestContext requestContext) {
     PurchaseOrder purchaseOrder = convertToPurchaseOrder(compPO);
-    Promise<List<PoLine>> promise = Promise.promise();
-
-    if (isEmpty(compPO.getCompositePoLines())) {
-      purchaseOrderLineService.getPoLinesByOrderId(compPO.getId(), requestContext)
-        .onSuccess(promise::complete)
-        .onFailure(promise::fail);
-    } else {
-      Future.succeededFuture()
-        .map(v -> {
-          List<PoLine> poLines = PoLineCommonUtil.convertToPoLines(compPO.getCompositePoLines());
-          if (initialOrdersStatus.equals(compPO.getWorkflowStatus().value()))
-            changeOrderStatus(purchaseOrder, poLines);
-          promise.complete(poLines);
-          return null;
-        })
-        .onFailure(promise::fail);
-    }
 
-    return promise.future()
+    return Future.succeededFuture()
+      .compose(v -> {
+        if (isEmpty(compPO.getCompositePoLines())) {
+          return purchaseOrderLineService.getPoLinesByOrderId(compPO.getId(), requestContext);
+        } else {
+          List<PoLine> poLines = PoLineCommonUtil.convertToPoLines(compPO.getCompositePoLines());
+          changeOrderStatusForOrderUpdate(purchaseOrder, poLines);
+          return Future.succeededFuture(poLines);
```

Comment:
a little confusion based on method name. if poLines exists in comPO we are chaning status but if not we are skipping. is it possible that fetched poLines have undesired status

Reply:
agree, it is not very obvious written initially, for example we use API to  GET composite order with lines and update workflowStatus for example to Closed with the same payload as we fetched from GET. In this case both order and lines will be included into PUT request.
But when UI is making PUT wiht Closed status it does not send po lines.
So decided to stop refactoring here for now to not introduce new issues.